tonic/src/codec/encode.rs

```diff
@@ -298,22 +298,21 @@ where
 }
 
 impl EncodeState {
-    fn trailers(&mut self) -> Result<Option<HeaderMap>, Status> {
+    fn trailers(&mut self) -> Option<Result<HeaderMap, Status>> {
         match self.role {
-            Role::Client => Ok(None),
+            Role::Client => None,
             Role::Server => {
                 if self.is_end_stream {
-                    return Ok(None);
+                    return None;
                 }
 
+                self.is_end_stream = true;
```

Comment:
This seems to be a change not related to upgrading, can you explain why this was moved out of the if statement?

Reply:
Ah I see let me know if this is correct, now that the data/trailers are combined into the same stateless poll fn we can always assume that when we call `trailers` that this is when the incoming stream of bytes has ended so the stream is always over at this point.

Reply:
Yeah, we only call `trailers` after the stream has returned `None`, so it is safe to assume it is done. Otherwise, we don't ever mark `is_end_stream` when the stream ends normally without an error.